jubatus/core/classifier/nearest_neighbor_classifier.cpp

```diff
@@ -100,12 +108,18 @@ void nearest_neighbor_classifier::train(
   }
   nearest_neighbor_engine_->set_row(id, fv);
   set_label(label);
+
+  util::concurrent::scoped_lock lk(label_mutex_);
+  labels_[label] += 1;
```

Comment:
Can't we use the same data structure (`jubatus::core::storage::labels`) in both NN-classifier and linear classifier?
Seems like doing a same thing twice.